core/engine.go

```diff
@@ -473,7 +510,7 @@ func (e *Engine) processSamples(sampleContainers []stats.SampleContainer) {
 
 	// TODO: run this and the below code in goroutines?
```

Comment:
Could we do that in this PR as we're at it?

Reply:
I'd prefer to do this as a separate PR, hopefully along the lines of https://github.com/grafana/k6/issues/2430

Reply:
Fair enough! 😸